api/v2beta1/helmrelease_types.go

```diff
@@ -32,6 +33,37 @@ import (
 const HelmReleaseKind = "HelmRelease"
 const HelmReleaseFinalizer = "finalizers.fluxcd.io"
 
+// PatcheJSON6902 JSON 6902 patches
+type PatcheJSON6902 struct {
+	// Patch is the YAML content of a patch.
+	Patch []apiextensionsv1.JSON `json:"patch,omitempty" yaml:"patch,omitempty"`
+
+	// Target points to the resources that the patch is applied to
+	Target kustypes.Selector `json:"target,omitempty" yaml:"target,omitempty"`
```

Comment:
We can not directly rely on the Kustomize API from within our own API, as it has been unstable at times. The kustomize-controller [works around this by duplicating types](https://github.com/fluxcd/kustomize-controller/blob/89017ca9cce6233b423ea657daff76d2178155bc/api/v1beta1/kustomization_types.go#L126-L139).

Reply:
OK, I refactored this by duplicating the types.